### ISSUE TITLE
Add the possibility to show the password

### ISSUE THREAD

**Issue by calixteman** [mozilla/pdf.js#15239]
Nowadays, it's usual for a user to be able to show a password.
For example, in the pdf viewer in Edge, there is the following modal (html):
![image](https://user-images.githubusercontent.com/5641725/181745349-7767c337-d8a0-4332-b965-8dd896b83b4d.png)

**Comment by Snuffleupagus** [comment]
This would (most likely) be trivial to implement, however we'd need Firefox designers to provide an appropriate icon for us to use.

**Comment by calixteman** [comment]
@rtestard, any thoughts ?

**Comment by Snuffleupagus** [comment]
A basic implementation, using the checkbox-approach: https://github.com/mozilla/pdf.js/compare/master...Snuffleupagus:pdf.js:password-show

**Comment by gregorypappas** [comment]
FWIW, Firefox already has a 'show password' button (pref'd off)
https://bugzilla.mozilla.org/show_bug.cgi?id=1743046

**Comment by Snuffleupagus** [comment]
> FWIW, Firefox already has a 'show password' button (pref'd off) https://bugzilla.mozilla.org/show_bug.cgi?id=1743046

Given this, it really seems to me that we should either:
 - Simply wait for that functionality to become enabled by default, and do nothing here.
 - Add a special-case, on the mozilla-central side, to enable that functionality *unconditionally* for code running at `resource://pdf.js/web/viewer.html`.

In any case, it seems somewhat questionable if we need/want to hack around this directly in the PDF.js viewer!?

**Comment by timvandermeij** [comment]
I agree, I think we should wait until that functionality is enabled upstream and not do changes for this here to not introduce unnecessary (temporary) complexity on the PDF.js side.

**Comment by Snuffleupagus** [comment]
@calixteman Do we perhaps want to do something similar to [bug 1790576](https://bugzilla.mozilla.org/show_bug.cgi?id=1790576) here as well, or should we just wontfix this?

**Comment by calixteman** [comment]
I enabled the pref `layout.forms.reveal-password-button.enabled` in nightly and I don't have any eye icon in the password input.
I didn't found any code using it:
https://searchfox.org/mozilla-central/search?q=reveal-password-button&path=&case=false&regexp=false
so I'd say that there's no implementation for the moment.

**Comment by Snuffleupagus** [comment]
> I enabled the pref `layout.forms.reveal-password-button.enabled` in nightly and I don't have any eye icon in the password input.

Strange, if works for me...
Note that you'll actually need to *type something first* into the password-input in order for the icon to show up though.

**Comment by calixteman** [comment]
It wfm either.... you're right about entering something first and about the capitalization either.

**Comment by Aditya-PS-05** [comment]
@calixteman , if this is not implemented. I would like to work on it. Can you assign me this issue?

**Comment by Snuffleupagus** [comment]
> if this is not implemented. I would like to work on it.

If you read https://github.com/mozilla/pdf.js/issues/15239#issuecomment-1229415119 it's hopefully clear that we haven't actually decided to fix this, and even if we do the code-changes would not be happening here anyway.
